src/styles/shared/components/buttons/_base.scss

```diff
@@ -0,0 +1,118 @@
+@use "sass:map";
```

Comment:
This is new: most mixins and functions private to `buttons` are in here. Most are direct c-n-p from `mixins.scss`. I'll highlight areas of change.